jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/vsdm/VsdmDepositor.java

```diff
@@ -0,0 +1,146 @@
+
+	private static Coder coder;
+	private String targetHost;
+	private int targetPort;
+	private int selfPort;
+	private DatagramSocket socket = null;
+
+	public VsdmDepositor() {
+		this(DEFAULT_TARGET_HOST, DEFAULT_TARGET_PORT, DEFAULT_SELF_PORT);
+	}
+
+	public VsdmDepositor(String targetHost, int targetPort, int selfPort) {
+		this.setTargetHost(targetHost);
+		this.setTargetPort(targetPort);
+		this.setSelfPort(selfPort);
+		coder = J2735.getPERUnalignedCoder();
+		try {
+			socket = new DatagramSocket(selfPort);
+			socket.setSoTimeout(DEFAULT_TIMEOUT);
+		} catch (SocketException e) {
+			logger.error("ODE: Error creating socket", e);
+		}
+	}
+
+	public String getTargetHost() {
+		return targetHost;
+	}
+
+	public void setTargetHost(String targetHost) {
+		this.targetHost = targetHost;
+	}
+
+	public int getTargetPort() {
+		return targetPort;
+	}
+
+	public void setTargetPort(int targetPort) {
+		this.targetPort = targetPort;
+	}
+
+	public int getSourcePort() {
+		return selfPort;
+	}
+
+	public void setSelfPort(int sourcePort) {
+		this.selfPort = sourcePort;
+	}
+
+	public void depositVsdm() {
+		logger.info("ODE: Initializing VSD deposit to SDC ...");
+		sendVsdServiceRequest();
+		receiveVsdServiceResponse();
+		sendVsdMessage();
+
+		if (socket != null){
+			logger.info("Closing VsdmDepositor Socket");
+			socket.close();
+		}	
+	}
+
+	private void sendVsdServiceRequest() {
+		ServiceRequest sr = CVSampleMessageBuilder.buildVehicleSituationDataServiceRequest();
+		ByteArrayOutputStream sink = new ByteArrayOutputStream();
+		try {
+			coder.encode(sr, sink);
+			byte[] payload = sink.toByteArray();
+			logger.info("ODE: Sending VSD Deposit ServiceRequest ...");
+			socket.send(new DatagramPacket(payload, payload.length, new InetSocketAddress(targetHost, targetPort)));
+		} catch (EncodeFailedException | EncodeNotSupportedException e) {
+			logger.error("ODE: Error Encoding VSD Deposit ServiceRequest", e);
+		} catch (IOException e) {
+			logger.error("ODE: Error Sending VSD Deposit ServiceRequest", e);
+		} 
+	}
+	
+	private void receiveVsdServiceResponse(){
+		try {
+			byte[] buffer = new byte[DEFAULT_BUFFER_LENGTH];
+			logger.info("ODE: Waiting for VSD deposit ServiceResponse...");
+			DatagramPacket responeDp = new DatagramPacket(buffer, buffer.length);
+			socket.receive(responeDp);
+
+			logger.info("ODE: Received VSD Deposit ServiceResponse");
+			if (buffer.length > 0) {
+				AbstractData response = J2735Util.decode(coder, buffer);
+				if (response instanceof ServiceResponse) {
+					logger.info("ODE: Printing VSD Deposit ServiceResponse {}", response.toString());
+				}
+			}
+		} catch (Exception e) {
+			logger.error("ODE: Failed to receive VSD Deposit ServiceResponse", e);
+		}
+	}
+	
+	private void sendVsdMessage(){
+		logger.info("ODE: Preparing VSD message deposit...");
+		VehSitDataMessage vsdm;
+		try {
+			vsdm = CVSampleMessageBuilder.buildVehSitDataMessage();
```

Comment:
Maybe we can get rid of this too, and just use the message we already encoded. 

(Remove the need for CVSampleMessageBuilder since it is just a helper class)